File: Python/isPowerOfThree.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n < 0:
            return False
        while (n > 1):
            if (n % 3 != 0):
                return False
            n /= 3
        return True

def main():
    solution = Solution()
    print(solution.isPowerOfThree(27))

if __name__ == "__main__":
    main()

        # The code provided is already complete and functional.
        # There is no need to add any additional code at the placeholder.

File: Newbie_basics/C++/isPowerOfThree.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution
{
private:
    /* data */
public:
    bool isPowerOfThree(int n){
        if (n <= 0) 
        return false;
        while (n>1)
        {
            if (n % 3 != 0) return false;
            n/=3;
        }
        return true;
    }

};

main()
{
    /* code */
    Solution solution;
    cout<< solution.isPowerOfThree(27)<<endl;
    return 0;
}

